diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -469,51 +484,44 @@ namespace diff_drive_controller
     odometry_.setMeasCovarianceParams(k_l_, k_r_);
 
     // COMPUTE AND PUBLISH ODOMETRY
-    if (open_loop_)
+    // Read wheel joint positions and velocities:
```

Comment:
I'm thinking of moving this (the reading) to the very top of the `update` method, so the values are readed the closest possible to the `time` the control cycle starts.

Now there's no problem reading before setting the dynamic params.